src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -675,21 +790,37 @@ export class ShadowGenerator implements IShadowGenerator {
      * @returns the Shadow Generator itself
      */
     public addShadowCaster(mesh: AbstractMesh, includeDescendants = true): ShadowGenerator {
-        if (!this._shadowMap) {
+        if (this.useCSM && this._cascadeShadowMaps) {
+            if (this._cascadeShadowMaps.length === 0) {

```

Comment:
not sure how we could end up in this case ?

Reply:
Fixed!